Lib/sqlite3/test/dbapi.py

```diff
@@ -724,6 +728,21 @@ def run(cur, errors):
         if len(errors) > 0:
             self.fail("\n".join(errors))
 
+    def test_dont_check_same_thread(self):
```

Comment:
Why this test is added in this PR?

Reply:
Because we lack tests for all keywords used in `sqlite3.connect` and `sqlite3.Connection.__init__`. See Berker's comment https://github.com/python/cpython/pull/24421#issuecomment-782840641